### ISSUE TITLE
Some details about this network.

### ISSUE THREAD

**Issue by huangxi6** [seoungwugoh/ivs-demo#5]
Hello, I'm interested in your work. And I have some questions about this network.
1. During pre-training, did you utilize the scribbles of the synthetic image data, too? Was the full network optimized?
2. During training, this article mentioned that "N is gradually increased from 4 to 8" and " the number of rounds also grows from 1 to 3", how to grow?
Thank you very much. Could you provide more code about the training process?

**Comment by seoungwugoh** [comment]
Hi, huangxi6.

1. Yes. You can see the major difference between pre-training and fine-tuning is the source of video frames. For pre-training, we synthesize them from static images. And, for fine-tuning, frames are sampled from real videos.
In addition, we use a small number of frames (N=2) and rounds (=2).

2. N grows by 1 at every 20 epoch. if N is lower than 7, the number of rounds is 2, else it is 3.
While our paper describes the number of rounds grows from 1 to 3, it actually grows from 2 to 3 due to some bugs (I just found them).
Sorry for providing incorrect information.

**Comment by huangxi6** [comment]
Thank you for your answers.
1. As for the first question, to my knowledge, the static images have no connection in Pascal VOC and other saliency datasets (as mentioned in your 2018 CVPR paper RGMP), why do you use N=2, rounds=2? In your paper, you described it as "the length of a training video clip". 

2. In addition, I want to know how many epochs do you use during pre-training or training.

3. About your synthetic scribbles, do you use the link "DAVIS Framework" (on the introduction page) to generate? Do you use several random selected scribbles from the video clip?

4. As for each video from 2017 DAVIS dataset, there are generally about 80 images. My question is that if you only take 4-8 images from each video wouldn't that leave out information. Maybe I misunderstood.
Looking forward to your reply.

**Comment by seoungwugoh** [comment]
Here are answers:
1. The parameters (N=2, rounds=2) are only for pre-training. 
That is the minimum number required for learning both functionalities: propagation and interaction. 
I don't understand other parts of question about our RGMP paper. Can you make it more clearer?

2. 280 epoches for pre-training (12000 samples per epoch). 70 epoches for main training (about 4000 sample per epoch).

3. We use similar way to DAVIS Framework, but we modified them for the speed-up.
Scribbles are generated on-the-fly according to the current state.  

4. Learning a deep networks using about 80 images is impractical due to GPU memory limits.
We validated model learned using 8 frames works fine when it is tested on 80-frame videos.

**Comment by huangxi6** [comment]
About N=2 and rounds=2, my question is that there are no two related images in these image segmentation datasets, such as Pascal VOC. If you use N=2, it means that you need two related images to train and propagate. So I don't understand. Do you use an original image with its variant (after crop or flip)?

**Comment by seoungwugoh** [comment]
We synthesize two related images from a single static image.
We used the same method with RGMP, please refer Sect 3.2 of our RGMP paper.

**Comment by huangxi6** [comment]
Thank you very much. The other question, do you use batchsize=1 during pre-training and training?

**Comment by seoungwugoh** [comment]
We used the largest batch size that allowed by GPU memory capacity.

**Comment by seoungwugoh** [comment]
Getting an empty mask seems to be due to bugs in the code.

**Comment by huangxi6** [comment]
Thank you for your answers. 
Best wishes.

**Comment by huangxi6** [comment]
Hello, I'm interested in the quantitative evaluation using DAVIS framework. And I try to evaluate your trained model, however, I can't gain your results. A strange fact is that the parameter 'max_nb_interactions' is set to 8, the real interactions are 24. Do you know what happened? Thank you!
